operator/controllers/lifecycle/keptntask/function_builder.go

```diff
@@ -0,0 +1,226 @@
+		}
+	}
+
+	params.Context = setupTaskContext(fb.options.task)
+
+	if len(fb.options.task.Spec.Parameters.Inline) > 0 {
+		err = mergo.Merge(&params.Parameters, fb.options.task.Spec.Parameters.Inline)
+		if err != nil {
+			controllercommon.RecordEvent(fb.options.recorder, apicommon.PhaseCreateTask, "Warning", fb.options.task, "TaskDefinitionMergeFailure", fmt.Sprintf("could not merge KeptnTaskDefinition: %s ", fb.options.task.Spec.TaskDefinition), "")
+			return nil, err
+		}
+	}
+
+	if fb.options.task.Spec.SecureParameters.Secret != "" {
+		params.SecureParameters = fb.options.task.Spec.SecureParameters.Secret
+	}
+	return &params, nil
+}
+
+func (fb *FunctionBuilder) parseFunctionTaskDefinition(spec *klcv1alpha3.RuntimeSpec, name string, namespace string, configMap string) (FunctionExecutionParams, bool, error) {
+	params := FunctionExecutionParams{}
+
+	// Firstly check if this task definition has a parent object
+	hasParent := false
+	if spec.FunctionReference != (klcv1alpha3.FunctionReference{}) {
+		hasParent = true
+	}
+	params.ConfigMap = configMap
+	if params.ConfigMap != "" && spec.HttpReference.Url != "" {
+		fb.options.Log.Info(fmt.Sprintf("The JobDefinition contains a ConfigMap and a HTTP Reference, ConfigMap is used / Namespace: %s, Name: %s  ", namespace, name))
+	}
+
+	// Check if there is a ConfigMap with the function for this object
+	if params.ConfigMap == "" {
+		// If not, check if it has an HTTP reference. If this is also not the case and the object has no parent, something is wrong
+		if spec.HttpReference.Url == "" && !hasParent {
+			return params, false, fmt.Errorf(controllererrors.ErrNoConfigMapMsg, namespace, name)
+		}
+		params.URL = spec.HttpReference.Url
+	}
+
+	// Check if there are parameters provided
+	if len(spec.Parameters.Inline) > 0 {
+		params.Parameters = spec.Parameters.Inline
+	}
+
+	// Check if there is a secret for secret params provided
+	if spec.SecureParameters.Secret != "" {
+		params.SecureParameters = spec.SecureParameters.Secret
+	}
+
+	// Check if there is a cmd params provided
+	if spec.CmdParameters != "" {
+		params.CmdParameters = spec.CmdParameters
+	}
+	return params, hasParent, nil
+}
+
+func (fb *FunctionBuilder) handleParent(ctx context.Context, params *FunctionExecutionParams) error {
+	var parentJobParams FunctionExecutionParams
+	parentDefinition, err := controllercommon.GetTaskDefinition(fb.options.Client, fb.options.Log, ctx, fb.options.funcSpec.FunctionReference.Name, fb.options.req.Namespace)
+	if err != nil {
+		controllercommon.RecordEvent(fb.options.recorder, apicommon.PhaseCreateTask, "Warning", fb.options.task, "TaskDefinitionNotFound", fmt.Sprintf("could not find KeptnTaskDefinition: %s ", fb.options.task.Spec.TaskDefinition), "")
+		return err
+	}
+	parSpec := controllercommon.GetRuntimeSpec(parentDefinition)
+	parentJobParams, _, err = fb.parseFunctionTaskDefinition(parSpec, parentDefinition.Name, parentDefinition.Namespace, parentDefinition.Status.Function.ConfigMap)
```

Comment:
the `hasParent` return value being ignored here means that we don't go further if the parent also only referes to another function, right? If yes, let's add a short comment here to make this clear